src/libs/tools/filter.c

```diff
@@ -371,477 +89,29 @@ void gui_init(dt_lib_module_t *self)
-
-  _lib_filter_update_query(user_data, DT_COLLECTION_PROP_RATING);
-}
-
-static void _lib_filter_sort_combobox_changed(GtkWidget *widget, gpointer user_data)
-{
-  /* update the ui last settings */
-  dt_collection_set_sort(darktable.collection, items[dt_bauhaus_combobox_get(widget)], -1);
-
-  /* save the images order */
-  _lib_filter_set_tag_order(user_data);
-
-  /* update the query and view */
-  _lib_filter_update_query(user_data, DT_COLLECTION_PROP_SORT);
-}
-
-static void _lib_filter_update_query(dt_lib_module_t *self, dt_collection_properties_t changed_property)
-{
-  /* sometimes changes */
-  dt_collection_set_query_flags(darktable.collection, COLLECTION_QUERY_FULL);
-
-  /* updates query */
-  dt_collection_update_query(darktable.collection, DT_COLLECTION_CHANGE_RELOAD, changed_property, NULL);
-}
-
-static void _reset_stars_filter(dt_lib_module_t *self, gboolean smart_filter)
-{
-  dt_lib_tool_filter_t *d = (dt_lib_tool_filter_t *)self->data;
-
-  if(smart_filter == TRUE)
-  {
-    /* initial import rating setting */
-    const int initial_rating = dt_conf_get_int("ui_last/import_initial_rating");
-
-    /* current selection in filter dropdown */
-    const int current_filter = dt_bauhaus_combobox_get(d->stars);
-
-    /* convert filter dropdown to rating: 2-6 is 1-5 stars, for anything else, assume 0 stars */
-    const int current_filter_rating = (current_filter >= 2 && current_filter <= 6) ? current_filter - 1 : 0;
-
-    /* new filter is the lesser of the initial rating and the current filter rating */
-    const int new_filter_rating = MIN(initial_rating, current_filter_rating);
-
-    /* convert new filter rating to filter dropdown selector */
-    const int new_filter = (new_filter_rating >= 1 && new_filter_rating <= 5) ? new_filter_rating + 1
-                                                                              : new_filter_rating;
-
-    /* Reset to new filter dropdown item */
-    dt_bauhaus_combobox_set(d->stars, new_filter);
-  }
-  else
-  {
-    /* Reset to topmost item, 'all' */
-    dt_bauhaus_combobox_set(d->stars, 0);
-  }
-}
-
-static void _lib_filter_reset(dt_lib_module_t *self, gboolean smart_filter)
-{
-  _reset_stars_filter(self, smart_filter);
-  _reset_text_filter(self);
-  _reset_colors_filter(self);
-}
-
-static gboolean _reset_filters(GtkAccelGroup *accel_group, GObject *acceleratable, guint keyval,
-                               GdkModifierType modifier, dt_lib_module_t *self)
-{
-  _lib_filter_reset(self, FALSE);
-  dt_collection_update_query(darktable.collection, DT_COLLECTION_CHANGE_RELOAD, DT_COLLECTION_PROP_SORT, NULL);
-  return TRUE;
-}
-
-void init_key_accels(dt_lib_module_t *self)
-{
-  dt_accel_register_lib(self, NC_("accel", "reset filters"), 0, 0);
-}
-
-void connect_key_accels(dt_lib_module_t *self)
-{
-  dt_accel_connect_lib(self, "reset filters", g_cclosure_new(G_CALLBACK(_reset_filters), self, NULL));
-}
-
-#ifdef USE_LUA
```

Comment:
I've not seen where the Lua support has been moved. Can you point it to me?

Reply:
husshh : I've dropped it entirely (but keep that for you, or I'll have complains -D)

More seriously, considering the amount of changes, I'll have to reimplement it from scratch in the filtering module. That was not really possible to reuse the code here...

Reply:
Ok, so we'll need help from @wpferguson on this. Right?

Reply:
I'll start looking at it

Reply:
:+1:  thanks a lot ! Just ask if you have any questions...